nomad/job_endpoint_hooks.go

```diff
@@ -134,6 +135,17 @@ var (
 		RTarget: cniMinVersion,
 		Operand: structs.ConstraintSemver,
 	}
+
+	// cniConsulConstraint is an implicit constraint added to jobs making use of
+	// transparent proxy mode.
+	cniConsulConstraint = &structs.Constraint{
+		LTarget: attrConsulCNI,
+		// TODO(tgross): consul-cni 1.4.2 will have Nomad support and will emit
+		// the correct version string. Once that's shipped update this RTarget
+		// to "1.4.2"
+		RTarget: "1.0.0",
```

Comment:
if 1.0.0 here wouldn't actually work, and 1.4.2 equally wouldn't work, why not go ahead and optimistically set it to 1.4.2?